roles/ceph-facts/tasks/facts.yml

```diff
@@ -269,3 +269,8 @@
   import_tasks: set_radosgw_address.yml
   when:
     - inventory_hostname in groups.get(rgw_group_name, [])
+
+- name: set_fact rgw_instances
+  set_fact:
+    rgw_instances: "{{ rgw_instances|default([]) | union([{'instance_name': 'rgw' + item|string, 'radosgw_address': _radosgw_address, 'radosgw_frontend_port': radosgw_frontend_port|int + item|int}]) }}"
```

Comment:
s/rgw_instances/radosgw_instances ?

the default value isn't needed anyway since you have a default in`ceph-facts/defaults/main.yml`

Reply:
em....the meanings of the 2 variables are different, the `radosgw_instances` is the number of rgw instances, default to 1, and `rgw_instances` is a list of key-value parameters of each rgw instance.
Here I want to induce the list `rgw_instances`.

Reply:
yes, I shall rename 1 of these 2 to make it clear.

Reply:
I think you are missing here a `when: inventory_hostname in groups.get(rgw_group_name, [])` on this task.

Reply:
`_radosgw_address` is set on rgw nodes. In the log I can see it undefined because you are running on a monitor node. I'm not sure it's the cause of your issue but it's still something you need to address.

Reply:
@guits oh, thanks for your reminding, these are my missing points, will fix these soon :)

Reply:
@guihecheng has this been addressed?

Reply:
@leseb yes, this is solved. As guits pointed, the case is the missing `when: inventory_hostname in groups.get(rgw_group_name, [])` that cause an mon host to set rgw_instances, so we got the original error.